src/componentManifest.ts

```diff
@@ -13,34 +13,155 @@ import { apiVersions } from './sdkVersion';
 const PLUGIN_NAME = 'componentManifest';
 const manifestPath = 'manifest.json';
 
+/**
+ * Descriptor for localized resources.
+ *
+ * On device, `locale` is capped at 64 bytes,
+ */
 interface Locales {
+  /**
+   * Lower case locale name.
+   *
+   * On device, this field is capped at 20 bytes (not including the null char).
+   * @example 'en-us'
+   */
   [locale: string]: {
+    /**
+     * Localized name of the app.
+     *
+     * On device, this field is capped at 30 bytes (not including the null
+     * char).
+     */
     name?: string;
+
+    /**
+     * Path to the resources file for the locale, relative to the manifest file.
+     *
+     * This file is generated from the corresponding '.po' file using an
+     * internal, more compact format. On device, this field is capped at
+     * 256 bytes (not including the null char).
+     * @example 'l/en-US'
+     */
     resources?: string;
   };
 }
+
 interface ComponentManifest {
+  /**
+   * API version.
+   *
+   * On device, this field is capped at 32 bytes (not including the null char).
+   */
   apiVersion: string;
+
+  /**
+   * Build ID, represented by a hex-formatted 64 bit int, prefixed with '0x'.
+   *
+   * On device, this field is capped at 20 bytes (not including the null char).
+   * @example '0xcafecafecafecafe'
+   */
   buildId: string;
+
+  /**
+   * The date the bundle was generated, formatted according to the ISO 8601
+   * standard.
+   */
   bundleDate: string;
+
+  /**
+   * Default non-localized app name.
+   *
+   * On device, this field is capped at 30 bytes (not including the null char).
+   */
   name: string;
+
+  /**
+   * List of permissions required by the app.
+   *
+   * On device, each permission name is capped at 30 bytes (not including the
+   * null char).
+   */
   requestedPermissions?: string[];
+
+  /**
+   * Unique app ID.
+   *
+   * On device, this field is capped at 36 bytes (not including the null char).
+   */
   uuid: string;
 }
 
+/**
+ * Valid application types.
+ *
+ * Enum values are hardcoded instead of reusing AppType since this
+ * interface describes the manifest as the device expects it,
+ * independent of data structures elsewhere. We want the build to
+ * fail if the AppType enum changes such that it becomes incompatible
+ * with what the device expects.
+ */
+type AppType = 'app' | 'clockface';
+
 interface DeviceManifest extends ComponentManifest {
+  /**
+   * The version of the manifest.
+   */
   appManifestVersion: 1;
-  // Enum values are hardcoded instead of reusing AppType since this
-  // interface describes the manifest as the device expects it,
-  // independent of data structures elsewhere. We want the build to
-  // fail if the AppType enum changes such that it becomes incompatible
-  // with what the device expects.
-  appType: 'app' | 'clockface';
+
+  /**
+   * JS App type.
+   *
+   * If none is specified, it will default to 'app' for backwards compatibility.
```

Comment:
This is the FW behaviour?

Reply:
Yes, for the current app manifest version.